walletconnect/main.js

```diff
@@ -0,0 +1,179 @@
+import {
+  AccountId,
+  TokenAssociateTransaction,
+  TokenId,
+} from "@hashgraph/sdk";
+import {
+  DAppConnector,
+  HederaJsonRpcMethod,
+  HederaSessionEvent,
+  HederaChainId,
+} from "@hashgraph/hedera-wallet-connect";
+
+// Global state for wallet connection
+let accountId = '';
+let isConnected = false;
+
+// Network configuration
+const NETWORK_CONFIG = {
+  testnet: {
+    network: "testnet",
+    jsonRpcUrl: "https://testnet.hedera.validationcloud.io/v1/GYSdi5Rlhc-NmoBLSVJGsqVQDOL6C4lCCQbyHc3NvsM",
+    mirrorNodeUrl: "https://testnet.mirrornode.hedera.com",
+    chainId: "0x128",
+  },
+};
+
+const walletConnectProjectId = "377d75bb6f86a2ffd427d032ff6ea7d3";
+const currentNetworkConfig = NETWORK_CONFIG.testnet;
+const hederaNetwork = currentNetworkConfig.network;
+const metadata = {
+  name: "Hedera Vanilla JS",
+  description: "Simple Hedera WalletConnect Integration",
+  url: window.location.origin,
+  icons: [window.location.origin + "/logo192.png"],
+};
+
+// Initialize WalletConnect
+const dappConnector = new DAppConnector(
+  metadata,
+  hederaNetwork,
+  walletConnectProjectId,
+  Object.values(HederaJsonRpcMethod),
```

Comment:
🤔 this seems brittle, because the imported object may contain additional keys/ or re-order its keys in future versions of the dependency. additionally the ECMAscript spec does not guarantee key order in `Object` too.

A less brittle approach would be to extract the needed values explicitly into an array literal.

not sure if the docs for `@hashgraph/hedera-wallet-connect` state that you should do this. if they do, then I guess this is OK.

Reply:
You need to use the values from `HederaJsonRpcMethod`. Currently, the transaction won't go through in this code snippet because of the wrong values here:

```ts
const jsonRpcMethods = [
  "getAccountBalance",  
  "getTransactionReceipt",
  "callSmartContract",
];
```
These are the values from HederaJsonRpcMethod.

```ts
export declare enum HederaJsonRpcMethod {
    GetNodeAddresses = "hedera_getNodeAddresses",// 1
    ExecuteTransaction = "hedera_executeTransaction",// 2
    SignMessage = "hedera_signMessage",// 3
    SignAndExecuteQuery = "hedera_signAndExecuteQuery",// 4
    SignAndExecuteTransaction = "hedera_signAndExecuteTransaction",// 5
    SignTransaction = "hedera_signTransaction"
}
```

You don't need to use Object.values() but it is less code and the order is not important.